querybook/webapp/components/DataDocQueryCell/DataDocQueryCell.tsx

```diff
@@ -446,40 +446,93 @@ class DataDocQueryCellComponent extends React.PureComponent<IProps, IState> {
                 sampleRate: this.sampleRate,
             },
         });
 
-        return runQuery(
-            await this.getTransformedQuery(),
-            this.engineId,
-            async (query, engineId) => {
-                const queryId = (
-                    await this.props.createQueryExecution(
-                        query,
-                        engineId,
-                        this.props.cellId,
-                        this.getQueryExecutionMetadata()
-                    )
-                ).id;
+        const transformedQuery = await this.getTransformedQuery();
+        const executionMetadata = this.getQueryExecutionMetadata();
+
+        try {
+            // Execute the query using the existing runQuery function
+            const queryId = await runQuery(
+                transformedQuery,
+                this.engineId,
+                async (query: string, engineId: number) => {
+                    const queryExecution =
+                        await this.props.createQueryExecution(
+                            query,
+                            engineId,
+                            this.props.cellId,
+                            executionMetadata,
+                            peerReviewParams,
+                            delayExecution
+                        );
+                    return queryExecution.id;
+                }
+            );
+
+            if (onSuccess) {
+                onSuccess(queryId);
+            }
+
+            return queryId;
+        } catch (error) {
+            toast.error('Failed to execute query.');
+            console.error('Query Execution Error:', error);
+            throw error;
+        }
+    }
 
+    @bind
+    public async onRunButtonClick() {
+        return this.executeQuery({
+            element: ElementType.RUN_QUERY_BUTTON,
+            onSuccess: (queryId: number) => {
                 // Only trigger survey if the query is modified within 5 minutes
                 if (Date.now() - this.state.modifiedAt < 5 * 60 * 1000) {
                     triggerSurvey(SurveySurfaceType.QUERY_AUTHORING, {
                         query_execution_id: queryId,
                         cell_id: this.props.cellId,
                     });
                 }
+            },
+        });
+    }
 
-                return queryId;
-            }
-        );
+    @bind
+    public async onPeerReviewSubmit(
+        reviewerIds: number[],
+        externalRecipients: string[],
+        notifierName: string,
+        justification: string
+    ) {
+        const peerReviewParams = {
+            reviewer_ids: reviewerIds,
+            external_recipients: externalRecipients,
+            notifier_name: notifierName,
+            review_request_reason: justification,
+        };
+
+        await this.executeQuery({
+            element: ElementType.PEER_REVIEW_QUERY_BUTTON,
+            peerReviewParams,
+            delayExecution: true,
```

Comment:
isn't `delayExecution` kind of unnecessary? it will be always true if peerReviewParams is not null?

Reply:
agreed, just added delayExecution flag in case we have future feature that require delay execution, removed for now